filesystem/file_manager.go

```diff
@@ -145,6 +146,7 @@ func (fm *FileManagement) iterateOverPackages(goVersionFileInfos []os.FileInfo)
 		fileName := file.Name()
 		versionNames = append(versionNames, fileName[:len(fileName)-len(fileExtraction)])
 	}
+	sort.SliceStable(versionNames, func(i, j int) bool { return versionNames[i] > versionNames[j] })
```

Comment:
It is not gonna sort as we expected. Could you write the input function in filesystem/version.go and write a unit test in filesystem/version_test.go for sorting?